Library/Homebrew/env_config.rb

```diff
@@ -15,6 +15,17 @@ module EnvConfig
         description: "Additional Google Analytics tracking ID to emit user behaviour analytics to. " \
                      "For more information, see: <https://docs.brew.sh/Analytics>",
       },
+      HOMEBREW_API_DOMAIN:                       {
+        description:  "Use this URL as the download mirror for Homebrew JSON API. " \
+                      "If metadata files at that URL are temporarily unavailable, " \
+                      "the default API domain will be used as a fallback mirror. " \
+                      "For example, `HOMEBREW_API_DOMAIN=http://localhost:8080` will cause " \
+                      "metadata files to download from the prefix `http://localhost:8080/`. " \
+                      "If files are not available at `HOMEBREW_API_DOMAIN` " \
+                      "they will be downloaded from the default API domain.",
```

Comment:
```suggestion
        description:  "Use this URL as the download mirror for Homebrew JSON API. " \
                      "If metadata files at that URL are temporarily unavailable, " \
                      "the default API domain will be used as a fallback mirror."
```

I don't think we need an example here. To me your description seems pretty complete